app/scripts/lib/message-manager.js

```diff
@@ -231,8 +246,12 @@ export default class MessageManager extends EventEmitter {
         event: reason,
         category: EVENT.CATEGORIES.TRANSACTIONS,
         properties: {
-          action: 'Sign Request',
+          action: 'Sign Request Reject',
           type: msg.type,
+          ui_customizations:
+            msg.securityProviderResponse?.flagAsDangerous === 1
+              ? ['flagged_as_malicious']
+              : [],
```

Comment:
I believe the ui_customization should be added to the events in createRPCMethodTrackingMiddleware versus adding a new event and renaming events here and in the other message managers.

Reply:
Hey @brad-decker, 
Do you have a suggestion how it should look like? I mean, in the middleware you mentioned I need the information I get from the message - `msg.securityProviderResponse?.flagAsDangerous`. 
Also, I have to mention the presence of the property that shares the same name as the one mentioned above (https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/createRPCMethodTrackingMiddleware.js#L173).

Reply:
You would add your value to the array already present. The events in these confirmations are not the ones we rely upon. Let me dig in a bit and see how you can get access to the value you need

Reply:
@brad-decker I tried it, but without success. The reason is that the data being passed to the `securityProviderRequest` from `createRPCMethodTrackingMiddleware` (`req`) is not proper, so as the response in that case. The proper data being sent is shown in the attached image (`personal-message-manager.js`).

![Data](https://user-images.githubusercontent.com/92310504/221200854-3bdf9965-d488-406c-a5bd-6a5231b3799a.png)

Reply:
@filipsekulic look at the detectSIWE implementation. Its done in both places as well. You will need to formulate the data object that securityProviderCheck needs -- note that in perosnal-message-manager that includes the 'id' of the transaction but the id key isn't used.

Reply:
@brad-decker I implemented the solution you suggested. Thanks! However, I could not fix the failing test - `createRPCMethodTrackingMiddleware.test.js`. I must be missing something. Hope you can help me. I would appreciate your help!